### ISSUE TITLE
[rancher-metadata] Can't run metadata stand-alone with provided example in README

### ISSUE THREAD

**Issue by rchrd** [rancher/rancher#8041]
**Rancher Versions:**
Server: n/a
healthcheck: n/a
ipsec: n/a
network-services: n/a
scheduler: n/a
kubernetes (if applicable):

**Docker Version:**
17.03.0-ce-rc1

**OS and where are the hosts located? (cloud, bare metal, etc):**
Docker for Mac

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
n/a

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
n/a

**Steps to Reproduce:**
Run the rancher-metadata container and mount the [example.json](https://github.com/rancher/rancher-metadata/blob/master/example/answers.json).

```
docker run --rm -v ${PWD}/answers.json:/tmp/answers.json rancher/metadata:v0.8.11 rancher-metadata --answers /tmp/answers.json
```

**Results:** 
The metadata service fails with an "unexpected EOF" message

```
time="2017-03-02T22:33:00Z" level=info msg="Starting rancher-metadata v0.8.11"
time="2017-03-02T22:33:00Z" level=info msg="Loading answers"
time="2017-03-02T22:33:00Z" level=error msg="Failed to load answers from file: unexpected EOF"
time="2017-03-02T22:33:00Z" level=fatal msg="Cannot startup without a valid Answers file"
```

**Comment by deniseschannon** [comment]
Docker for Mac isn't supported. http://docs.rancher.com/rancher/v1.5/en/installing-rancher/installing-server/#requirements

**Comment by rchrd** [comment]
Exactly the same problem occurs on:

**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 16.04.2 LTS on VirtualBox

**Docker Version**
1.12.6
1.13.1
17.03.0-ce

**Comment by yevheniidehtiar** [comment]
**why it issue is closed?**
I have same problem

**OS** 
Ubuntu 16.04

**Docker Version**
1.12.6

**Comment by slavniyteo** [comment]
I use pure bin (compiled with go from repository, commit ac5caa4 from 17 Aug) and got the same error.

**OS:** Ubuntu 16.04

**answers.json** (from [Readme.md](https://github.com/rancher/rancher-metadata#json)):
```json
{
  "2015-12-19": {
    "10.1.2.2": {
      "key1": "value1",
      "arbitrarily": {
        "nested": [
          "JSON", "of", "any", "type", 42, null, false
        ]
      }
    },

    "192.168.0.2": {
      "key1": "value2"
    },

    "default": {
      "key1": "value3"
    }
  },

  "2015-07-25": {
  }
}
```

**Comment by michaellopez** [comment]
I'm seeing this too with v0.9.5.

**Comment by michaellopez** [comment]
I tracked this down to [this commit](https://github.com/rancher/rancher-metadata/commit/b144367659b65f7bdbad40a8d37905105b2615f6#diff-e871404fd69cab40521d6b2178847cf7). It was released in `v0.8.0`. Basically the answers file are being encoded and also later compressed, meaning `rancher-metadata` expects a encoded and compressed answers file, which one usually does not want to provide in stand-alone mode.

My workaround right now is to use `v0.7.4` until `rancher-metadata` gets fixed and can detect an unencoded, uncompressed answers file on startup.
